tests/test_api_lib_policy.py

```diff
@@ -1596,7 +1598,16 @@ def test_06_offline_auth(self):
         self.assertEqual(len(response), 100)
         # check if the counter of the token was increased to 100
         tokenobject = get_tokens(serial=serial)[0]
-        self.assertEqual(tokenobject.token.count, 101)
```

Comment:
I think the count should be at 100 rather than 101 here:
We use a fresh token which is created with `count = 0`. Then, we obtain 100 offline OTPs. Thus, the resulting `count` value should be 100.

Reply:
The token.count in the database is the next value, that the user is allowed to use! Not the last value, which was used by the user!

I.e. if the ``token.count`` is  set ot 0 initially, then the user can use count=0 and  ``token.count`` will be set to 1.

So if the offline token issues 100 values... are these the values 0..99? 
Then the last value, which the user can use is the value 99 and if he does so the ``token.count`` will be set in the database to 100. 100 would be the next OTP value, which would be accepted, but which is **not** contained on the list.

You can check this with the PAPER token.
I am not sure, why it is set to 101, it should happen here:
https://github.com/privacyidea/privacyidea/pull/842/files#diff-c32ef9ce1beb3a0a39d033ecdc356ca0R148

Reply:
Oops yes, this testcase is confusing: In reality, the user uses `count=0` (i.e. OTP "287082") to request 100 offline OTPs, so the counter would indeed be `count=101` after that.
However, this testcase does not actually invoke the ``/validate/check`` endpoint, but only the ``offline_info`` function, so "287082" is not actually consumed.
I should probably extend the ``test_12_auth_items_offline`` testcase in ``test_api_machines.py`` so that we have a real API-level testcase for the offline OTP.